From 10bc982dfd389a82b014312b67b45ce10a6831d6 Mon Sep 17 00:00:00 2001
From: Sergio Durigan Junior <sergiodj@chainguard.dev>
Date: Wed, 18 Feb 2026 18:06:05 -0500
Subject: [PATCH] Unset GOFIPS140 during bootstrap

This is needed because otherwise the build fails due to the use of
"purego".

See
https://github.com/golang/go/commit/8105d0ccc273afa717ba536f4d42dac3920c017e
for more information.

Signed-off-by: Sergio Durigan Junior <sergiodj@chainguard.dev>
---
 src/cmd/dist/buildtool.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cmd/dist/buildtool.go b/src/cmd/dist/buildtool.go
index 62cd937692..70ddd620b5 100644
--- a/src/cmd/dist/buildtool.go
+++ b/src/cmd/dist/buildtool.go
@@ -232,6 +232,11 @@ func bootstrapBuildTools() {
 	defer os.Setenv("GOBIN", os.Getenv("GOBIN"))
 	os.Setenv("GOBIN", "")
 
+	// Chainguard: We have to unset GOFIPS140 here otherwise the build
+	// fails due to the use of "purego".
+	defer os.Setenv("GOFIPS140", os.Getenv("GOFIPS140"))
+	os.Setenv("GOFIPS140", "")
+
 	os.Setenv("GOOS", "")
 	os.Setenv("GOHOSTOS", "")
 	os.Setenv("GOARCH", "")
-- 
2.51.0

